core/src/MsalIdHelper.cs

```diff
@@ -35,83 +35,103 @@ namespace Microsoft.Identity.Core
+                if (platformProxy == null)
+                {
+                    throw CoreExceptionFactory.Instance.GetClientException(
+                        CoreErrorCodes.PlatformNotSupported,
+                        CoreErrorMessages.PlatformNotSupported);
+                }
 
-            var processorInformation = platformProxy.GetProcessorArchitecture();
-            if (processorInformation != null)
-            {
-                parameters[MsalIdParameter.CpuPlatform] = processorInformation;
-            }
+                var parameters = new Dictionary<string, string>
+                {
+                    [MsalIdParameter.Product] = platformProxy.GetProductName(),
+                    [MsalIdParameter.Version] = GetMsalVersion()
+                };
 
-            var osInformation = platformProxy.GetOperatingSystem();
-            if (osInformation != null)
-            {
-                parameters[MsalIdParameter.OS] = osInformation;
-            }
+                string processorInformation = platformProxy.GetProcessorArchitecture();
+                if (processorInformation != null)
+                {
+                    parameters[MsalIdParameter.CpuPlatform] = processorInformation;
+                }
+
+                string osInformation = platformProxy.GetOperatingSystem();
+                if (osInformation != null)
+                {
+                    parameters[MsalIdParameter.OS] = osInformation;
+                }
+
+                string deviceInformation = platformProxy.GetDeviceModel();
+                if (deviceInformation != null)
+                {
+                    parameters[MsalIdParameter.DeviceModel] = deviceInformation;
+                }
+
+                return parameters;
+            });
 
-            var deviceInformation = platformProxy.GetDeviceModel();
-            if (deviceInformation != null)
+        private static readonly Lazy<string> MsalVersion = new Lazy<string>(
+            () =>
             {
-                parameters[MsalIdParameter.DeviceModel] = deviceInformation;
-            }
+                string fullVersion = typeof(MsalIdHelper).GetTypeInfo().Assembly.FullName;
+                var regex = new Regex(@"Version=[\d]+.[\d+]+.[\d]+.[\d]+");
+                var match = regex.Match(fullVersion);
+                if (!match.Success)
+                {
+                    return null;
+                }
 
-            return parameters;
+                string[] version = match.Groups[0].Value.Split(
+                    new[]
+                    {
+                        '='
+                    },
+                    StringSplitOptions.None);
+                return version[1];
+            });
+
+        public static IDictionary<string, string> GetMsalIdParameters()
+        {
+            return MsalIdParameters.Value;
         }
 
         public static string GetMsalVersion()
```

Comment:
>GetMsalVersion [](start = 29, length = 14)

I suppose putting this in core will cause challenges? Ideally I would have liked to see this one in the PlatformProxy or similar.

Reply:
There are several things that can move into PlatformProxy but will screw up ADAL or potentially bring in code into ADAL that it doesn't want/need.  I'm holding on those until after "the split(tm)"